packages/core/admin/admin/src/services/users.ts

```diff
@@ -0,0 +1,92 @@
+import { FindRole, FindRoles } from '../../../shared/contracts/roles';
+import * as Users from '../../../shared/contracts/user';
+
+import { adminApi } from './api';
+
+const usersService = adminApi.injectEndpoints({
+  endpoints: (builder) => ({
+    /**
+     * users
+     */
+    updateUser: builder.mutation<
+      Users.Update.Response['data'],
+      Omit<Users.Update.Request['body'] & Users.Update.Params, 'blocked'>
+    >({
+      query: ({ id, ...body }) => ({
+        url: `/admin/users/${id}`,
+        method: 'PUT',
+        data: body,
+      }),
+      invalidatesTags: (_res, _err, { id }) => [{ type: 'User', id }],
+    }),
+    getUsers: builder.query<
```

Comment:
Do you think we should add provideTags for this, so when user is invalidated we refetch this?

Reply:
Yeah I think I've done this change locally now! A lot of the work is incremental on the branch so I'm finding more places I want to invalidate data - great suggestion nonetheless though!